pallets/stellar-relay/src/lib.rs

```diff
@@ -576,7 +577,21 @@ pub mod pallet {
 				.ok_or(Error::<T>::MissingExternalizedMessage)?;
 
 			// Variable used to check if all envelopes are using the same slot index
-			let slot_index: u64 = externalized_envelope.statement.slot_index;
+			let slot_index = externalized_envelope.statement.slot_index;
+
+			// We store the externalized value in a variable so that we can check if it's the same
+			// for all envelopes. We don't distinguish between externalized and confirmed values as
+			// it should be the same value regardless.
+			let (externalized_value, mut externalized_n_h) =
+				match &externalized_envelope.statement.pledges {
+					ScpStatementPledges::ScpStExternalize(externalized_statement) =>
+						(&externalized_statement.commit.value, externalized_statement.n_h),
+					_ => return Err(Error::<T>::ExternalizedValueNotFound),
```

Comment:
Wouldn't it be enough to have just one of the errors `MissingExternalizedMessage` and `ExternalizedValueNotFound`? They are basically the same (meaning that there is no externalized message).